### PR TITLE
Refactor RPs for new GetKey API #1 - [MOD-5256]

### PR DESCRIPTION
**Pull request by GuyAv46**
changes:
1. Sorter will no longer throw away docs with missing data but will give them a lower rank than any other doc, so it will return an empty doc only if there are no better documents that passed the filter. This is aligned with the changes presented in #2651

### CODE REVIEW

File: src/result_processor.c [hunk 1]
```diff
@@ -684,35 +649,25 @@ static int rploaderNext(ResultProcessor *base, SearchResult *r) {
...
-    loadopts.mode |= RLOOKUP_LOAD_ALLKEYS;
-  }
+
+  lc->loadopts.sctx = lc->base.parent->sctx; // TODO: can be set in the constructor
+  lc->loadopts.dmd = r->dmd;
+
   // if loading the document has failed, we return an empty array
-  RLookup_LoadDocument(lc->lk, &r->rowdata, &loadopts);
+  if (RLookup_LoadDocument(lc->lk, &r->rowdata, &lc->loadopts) != REDISMODULE_OK) {
+    QueryError_ClearError(&lc->status);
```
Comment (DvirDukhan) [inline src/result_processor.c:663]: Add a comment that error code is ignored

File: src/aggregate/aggregate_request.c [hunk 2]
```diff
@@ -1203,19 +1203,20 @@ int buildOutputPipeline(AREQ *req, QueryError *status) {
   // Add a LOAD step...
   const RLookupKey **loadkeys = NULL;
   if (req->outFields.explicitReturn) {
-    bool is_old_json = isSpecJson(req->sctx->spec) && (req->reqConfig.dialectVersion < APIVERSION_RETURN_MULTI_CMP_FIRST);
     // Go through all the fields and ensure that each one exists in the lookup stage
+    uint32_t flags = RLOOKUP_F_EXPLICITRETURN;
+    // If we have a HASH spec or a "new" JSON API version (DIALECT), we simply load all requested and missing fields.
```
Comment (DvirDukhan) [inline src/aggregate/aggregate_request.c:1208]:
```suggestion
    // If we have a HASH spec or a "new" JSON API version (DIALECT >= 3), we simply load all requested and missing fields.
```

File: src/result_processor.c (hunk 1 above)
Comment (GuyAv46) [reply src/result_processor.c:663]: Fixed on `LOADER` PR

File: src/rlookup.c [hunk 3]
```diff
@@ -129,24 +131,34 @@ RLookupKey *RLookup_GetKeyEx(RLookup *lookup, const char *name, size_t name_len,
...
       setKeyByFieldSpec(key, fs);
-      if (key->flags & RLOOKUP_F_ISLOADED) {
-        // If the key is marked as loaded, it means that it is sortable and un-normalized,
+      if (key->flags & RLOOKUP_F_VAL_AVAILABLE && !(flags & RLOOKUP_F_FORCE_LOAD)) {
+        // If the key is marked as "value available", it means that it is sortable and un-normalized.
         // so we can use the sorting vector as the source, and we don't need to load it from the document.
         return NULL;
       }
+    } else {
+      key->path = field_name;
```
[the remark this answers is not in the document]
Comment (GuyAv46) [reply src/rlookup.c:161]: modified at the `LOADER` PR. Fixed there

**Review by DvirDukhan — APPROVED** [review]

**Pull request merged**